File: XPSC/weekly Task/week 8/day 3/A_Bachgold_Problem.cpp
#include <bits/stdc++.h>

#define br "\n"
#define ll long long
#define yes cout << "YES\n"
#define no cout << "NO\n"
using namespace std;

class Solution
{
public:
    void solve()
    {
        ll n;
        cin >> n;
        ll ans = 0;
        if (n % 2 == 0)
        {
            ans = n / 2;
            cout << ans << br;
            ll i = 0;
            while (i < ans)
            {
                cout << 2 << " ";
                i++;
            }
            cout << br;
        }
        else
        {
            n -= 3;
            ans = n / 2;
            cout << ans + 1 << br;
            ll i = 0;
            while (i < ans)
            {
                cout << 2 << " ";
                i++;
            }
            cout << 3 << br;
        }
    }
};
int main()
{
    // _Created :  13 May 2024 ||   22:08:41
    // _File    :  A_Bachgold_Problem.cpp
    // Writer    :  Md. Nazmus Sakib(engrsakib)
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    // اللهم صل على سيدنا محمدن النبى الامى وآله وسلم
    ios::sync_with_stdio(false);
    cin.tie(NULL);
    ll t = 1;
    // cin >> t;
    while (t--)
    {
        Solution obj;
        obj.solve();
    }
    return 0;
}

File: XPSC/weekly Task/week 8/day 3/A_Bachgold_Problem.java
import java.util.Scanner;

public class A_Bachgold_Problem {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        Solution obj = new Solution();

        long t = 1;
        // long t = sc.nextLong();
        while (t-- > 0) {
            obj.solve();
        }
        sc.close();
    }
}

class Solution {
    public void solve() {
        Scanner sc = new Scanner(System.in);
        long n = sc.nextLong();
        long ans = 0;

        if (n % 2 == 0) {
            ans = n / 2;
            System.out.println(ans);
            for (int i = 0; i < ans; i++) {
                System.out.print("2 ");
            }
            System.out.println();
        } else {
            n -= 3;
            ans = n / 2;
            System.out.println(ans + 1);
            for (int i = 0; i < ans; i++) {
                System.out.print("2 ");
            }
            System.out.println("3");
        }
        sc.close();
    }
}
